Gems/ScriptCanvas/Code/Editor/View/Windows/MainWindow.cpp

```diff
@@ -1145,8 +1147,16 @@ namespace ScriptCanvasEditor
         AZ_Assert(slot, "A valid slot must be provided");
         if (slot)
         {
+            
             m_slotTypeSelector = new SlotTypeSelectorWidget(GetActiveScriptCanvasId(), this); // Recreate the widget every time because of https://bugreports.qt.io/browse/QTBUG-76509
-            m_slotTypeSelector->PopulateVariablePalette(m_variablePaletteTypes);
+            if (azrtti_istypeof<const ScriptCanvas::Nodes::Math::MathExpression*>(slot->GetNode()))
+            {
+                m_slotTypeSelector->PopulateVariablePalette(m_variablePaletteTypes,true);
+            }
+            else
+            {
+                m_slotTypeSelector->PopulateVariablePalette(m_variablePaletteTypes);
+            }
```

Comment:
This is not a scalable pattern, the key problem is that it establishes precedent for future changes to expand on this "if it's this node, do that", the code becomes difficult to maintain and bloated. Instead, I recommend to consider a way for the slot type selector's behavior to be configurable or specialized for the type of node it's operating on. There are different ways this could be done, but the goal should be for it to be easy and very clean to add a new rule for the slot type selector's behavior without changing external code.